### PR TITLE
WL-2482

### PR DESCRIPTION
**Pull request by buckett**
Changes that create conversion for existing content in plain text forums, but also improve the filtering we apply to each post.

### CODE REVIEW

File: messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/jobs/UpdateMarkupFreeContent.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ * @author buckett
+ *
+ */
+public class UpdateMarkupFreeContent implements Job {
+
+	private static final Log LOG = LogFactory.getLog(UpdateMarkupFreeContent.class);
+	
+	// These are escaped tags which should have been stripped.
+	// Use non-greedy matcher.
+	private static final Pattern anchorTags = Pattern.compile("<a.*?href=('|\")(.*?)\\1.*?>(.*?)</a>");
```
Comment (ColinHebert) [inline messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/jobs/UpdateMarkupFreeContent.java]: This regular expression could also capture wrong anchors.

`<a title="hello">blah blah</a><a href="test">test</a>` will be captured only once and will probably result in `<a href="test">test</a>`.

Could that happen?

File: messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/jobs/UpdateMarkupFreeContent.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			}
+		}
+		anchorMatcher.appendTail(anchorBuilder);
+		
+		String newBody = anchorBuilder.toString().replaceAll("\n", "<br />");
+		newBody = newBody.replaceAll("</div>", "<br /><br />");
+
+		String clean = messageParsingService.parse(messageParsingService.format(newBody));
+		// Trim multiple <br />
+		clean = clean.replaceAll("<br />(:?\\w*<br />)+", "<br /><br />");
```
Comment (ColinHebert) [inline messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/jobs/UpdateMarkupFreeContent.java]: Shouldn't it be `\W` and even so wouldn't `\b` or `\s` be more appropriate?

**Comment by buckett** [comment]
Thanks, updated code and added tests.

File: messageforums-component-impl/src/test/org/sakaiproject/component/app/messageforums/TestMessageParsingServiceImpl.java [hunk 3]
```diff
@@ -62,6 +62,127 @@ public void test4() {
...
+	@Test
+	public void testInBrackets() {
+		assertEquals("(<a href='http://www.ox.ac.uk' target='_blank'>http://www.ox.ac.uk</a>)", messageParsingService.parse("(http://www.ox.ac.uk)"));
+		assertEquals("(<a href='http://www.cam.ac.uk/' target='_blank'>http://www.cam.ac.uk/</a>)", messageParsingService.parse("(http://www.cam.ac.uk/)"));
+		assertEquals("(<a href='http://www.ex.ac.uk' target='_blank'>www.ex.ac.uk</a>)", messageParsingService.parse("(www.ex.ac.uk)"));
+		assertEquals("(<a href='https://itunes.apple.com/gb/id288751446?mt=8' target='_blank'>https://itunes.apple.com/gb/id288751446?mt=8</a>)",
+				messageParsingService.parse("(https://itunes.apple.com/gb/id288751446?mt=8)"));
+	}
+	
+	//@Test
```
Comment (ColinHebert) [inline messageforums-component-impl/src/test/org/sakaiproject/component/app/messageforums/TestMessageParsingServiceImpl.java]: Why is this test commented?

Comment (buckett) [inline messageforums-component-impl/src/test/org/sakaiproject/component/app/messageforums/TestMessageParsingServiceImpl.java]: Added reason for test not being run.

**Comment by ColinHebert** [comment]
Merged

**Pull request merged**